controllers/quay/features.go

```diff
@@ -35,267 +33,311 @@ const (
+		)
 	}
 
-	r.Log.Info("cluster does not support `Route` API", "error", err)
+	wildcard, err := getCertificatesPEM(rt.Spec.Host + ":443")
+	if err != nil {
+		return err
+	}
+	qctx.ClusterWildcardCert = wildcard
 
-	return ctx, quay, nil
+	return r.Client.Delete(ctx, &rt)
 }
 
-func (r *QuayRegistryReconciler) checkObjectBucketClaimsAvailable(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, configBundle map[string][]byte) (*quaycontext.QuayRegistryContext, *v1.QuayRegistry, error) {
-	datastoreName := types.NamespacedName{Namespace: quay.GetNamespace(), Name: quay.GetName() + "-quay-datastore"}
-	var objectBucketClaims objectbucket.ObjectBucketClaimList
-	if err := r.Client.List(context.Background(), &objectBucketClaims); err == nil {
-		r.Log.Info("cluster supports `ObjectBucketClaims` API")
-
-		ctx.SupportsObjectStorage = true
-
-		found := false
-		for _, obc := range objectBucketClaims.Items {
-			if obc.GetNamespace()+"/"+obc.GetName() == datastoreName.String() {
-				found = true
-				r.Log.Info("`ObjectBucketClaim` exists")
-
-				var datastoreSecret corev1.Secret
-				if err = r.Client.Get(context.Background(), datastoreName, &datastoreSecret); err != nil {
-					r.Log.Error(err, "unable to retrieve Quay datastore `Secret`")
+func (r *QuayRegistryReconciler) checkObjectBucketClaimsAvailable(
+	ctx context.Context, qctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry,
+) error {
+	dstorensn := types.NamespacedName{
+		Name:      fmt.Sprintf("%s-quay-datastore", quay.GetName()),
+		Namespace: quay.GetNamespace(),
+	}
 
-					return ctx, quay, err
-				}
+	var claims objectbucket.ObjectBucketClaimList
+	if err := r.Client.List(ctx, &claims); err != nil {
+		r.Log.Info("cluster does not support `ObjectBucketClaim` API")
+		return nil
+	}
 
-				var datastoreConfig corev1.ConfigMap
-				if err = r.Client.Get(context.Background(), datastoreName, &datastoreConfig); err != nil {
-					r.Log.Error(err, "unable to retrieve Quay datastore `ConfigMap`")
+	qctx.SupportsObjectStorage = true
+	r.Log.Info("cluster supports `ObjectBucketClaims` API")
 
-					return ctx, quay, err
-				}
+	for _, obc := range claims.Items {
+		if obc.GetNamespace()+"/"+obc.GetName() != dstorensn.String() {
+			continue
+		}
 
-				r.Log.Info("found `ObjectBucketClaim` and credentials `Secret`, `ConfigMap`")
+		r.Log.Info("`ObjectBucketClaim` exists")
 
-				host := string(datastoreConfig.Data[datastoreBucketHostKey])
-				if strings.Contains(host, ".svc") && !strings.Contains(host, ".svc.cluster.local") {
-					r.Log.Info("`ObjectBucketClaim` is using in-cluster endpoint, ensuring we use the fully qualified domain name")
-					host = strings.ReplaceAll(host, ".svc", ".svc.cluster.local")
-				}
+		var datastoreSecret corev1.Secret
+		if err := r.Client.Get(ctx, dstorensn, &datastoreSecret); err != nil {
+			r.Log.Error(err, "unable to retrieve Quay datastore `Secret`")
+			return fmt.Errorf("awaiting for bucket claim to be processed")
```

Comment:
Should we also log this message on line 229? `"unable to retrieve Quay datastore Secret"` is quite cryptic on its own...

Reply:
I don't follow, what do you mean?

Reply:
When this happens, the logs will show an error saying "unable to retrieve Quay datastore `Secret`". That alone doesn't say much.
My suggestion would be to add something like the returned error's message to the log, i.e:
```
r.Log.Error(err, "unable to retrieve Quay datastore `Secret` - bucket claim not ready")
```
Or something in the likes.

Reply:
Fixed, PTAL.